downloader/src/main/kotlin/vcs/Git.kt

```diff
@@ -22,7 +22,7 @@ package com.here.ort.downloader.vcs
 import com.here.ort.downloader.WorkingTree
 import com.here.ort.model.VcsInfo
 import com.here.ort.model.VcsType
-import com.here.ort.utils.LicenseFileMatcher
+import com.here.ort.utils.FileMatcher
```

Comment:
Commit message: How does renaming on its own increase reusability? I guess what you're meaning to say is that a more generic name is more suitable now, as the `DEFAULT_NAMES` property is gone thus the code does  not suggest anymore to be only useful for *license* files.

Anyway, to be hosted this whole commit series did not make much sense to me until this last commit, which finally revealed the real goal: Refactor `LicenseFileMatcher` into a more generic `FileMatcher` that is not necessarily limited to license files. I'd suggest to squash all commits in this PR and use something similar like my last sentence as the commit message title.

Reply:
Done.